### PR TITLE
PMM-10384 Handle example query placeholders

### PR DESCRIPTION
**Pull request by matejkubinec**

### CODE REVIEW

File: pmm-app/src/pmm-qan/panel/components/Details/Example/Example.tools.tsx [hunk 1]
```diff
@@ -22,3 +23,11 @@ export const getExample = (databaseType) => (example: any): any => {
     </Highlight>
   );
 };
+
+export const getExampleItem = (databaseType: DatabasesType, example: QueryExampleResponseItem) => {
+  if (databaseType === 'mysql') {
```
[the remark this answers is not in the document]
Comment (matejkubinec) [reply pmm-app/src/pmm-qan/panel/components/Details/Example/Example.tools.tsx:28]: It is a string union type.

File: pmm-app/src/pmm-qan/panel/components/Details/Explain/Explain.tools.ts [hunk 2]
```diff
@@ -52,3 +61,60 @@ export const parseExplain = (result: ActionResult) => {
...
+  try {
+    if (databaseType === Databases.mysql && (placeholders || !example.placeholders_count)) {
+      const payload = {
+        example,
+        queryId,
+        placeholders,
+      };
+
+      const traditionalExplainActionId = await mysqlMethods.getExplainTraditional(payload);
+      const jsonExplainActionId = await mysqlMethods.getExplainJSON(payload);
```
[the remark this answers is not in the document]
Comment (matejkubinec) [reply pmm-app/src/pmm-qan/panel/components/Details/Explain/Explain.tools.ts:80]: Good point, I'll refactor it.

File: pmm-app/src/pmm-qan/panel/components/Details/Explain/Explain.tools.ts [hunk 3]
```diff
@@ -52,3 +61,60 @@ export const parseExplain = (result: ActionResult) => {
...
+
+      return {
+        jsonExplain: jsonResult,
+        classicExplain: actionResult,
+        visualExplain: actionResult,
+      };
+    }
+
+    return {
+      jsonExplain: actionResult,
```
[the remark this answers is not in the document]
Comment (matejkubinec) [reply pmm-app/src/pmm-qan/panel/components/Details/Explain/Explain.tools.ts:105]: Be default, if its an unsupported database type/explain/etc. the default actionResult is returned.

**Review by fabio-silva — APPROVED** [review]

**Review by solovevayaroslavna — DISMISSED** [review]

**Review by fabio-silva — APPROVED** [review]

File: pmm-app/src/pmm-qan/panel/components/Details/ExplainPlaceholders/PlaceholdersForm/PlaceholdersForm.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          <div className={styles.container}>
+            <div className={styles.follow}>{Messages.follow}</div>
+            <PrepareExplainFingerPrint
+              placeholders={values.placeholders}
+              fingerprint={example.explain_fingerprint || ''}
+            />
+            <p className={styles.description}>{Messages.replace}</p>
+            <div>
+              {placeholders?.map((placeholder) => (
+                <FinalField
```
Comment (solovevayaroslavna) [inline pmm-app/src/pmm-qan/panel/components/Details/ExplainPlaceholders/PlaceholdersForm/PlaceholdersForm.tsx:33]: what about keys there?

File: pmm-app/src/pmm-qan/panel/components/Details/ExplainPlaceholders/PlaceholdersForm/PlaceholdersForm.utils.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { times } from 'lodash';
+import { PlaceholderInput } from './PlaceholdersForm.types';
+
+export const prepareInputs = (count: number): PlaceholderInput[] => times(count, (idx) => ({
+  label: `:${idx + 1}`,
```
Comment (solovevayaroslavna) [inline pmm-app/src/pmm-qan/panel/components/Details/ExplainPlaceholders/PlaceholdersForm/PlaceholdersForm.utils.ts:5]: why +1 ?

File: pmm-app/src/pmm-qan/panel/components/Details/ExplainPlaceholders/QueryFingerprint/QueryFingerprint.utils.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import sqlFormatter from 'sql-formatter';
+
+export const replacePlaceholders = (fingerprint: string, placeholders: string[] = []): string => {
+  let replaced = fingerprint || '';
+
+  placeholders.forEach((value, idx) => {
+    const fingerprintIdx = idx + 1;
```
Comment (solovevayaroslavna) [inline pmm-app/src/pmm-qan/panel/components/Details/ExplainPlaceholders/QueryFingerprint/QueryFingerprint.utils.ts:7]: the same question, don't understend now what's the magic number and why we need to add this.

**Review by solovevayaroslavna — APPROVED** [review]

File: pmm-app/src/pmm-qan/panel/components/Details/ExplainPlaceholders/PlaceholdersForm/PlaceholdersForm.tsx (hunk 4 above)
Comment (solovevayaroslavna) [reply pmm-app/src/pmm-qan/panel/components/Details/ExplainPlaceholders/PlaceholdersForm/PlaceholdersForm.tsx:33]: hmm, react takes indexes by default, so it's probably not necessary here

Comment (matejkubinec) [reply pmm-app/src/pmm-qan/panel/components/Details/ExplainPlaceholders/PlaceholdersForm/PlaceholdersForm.tsx:33]: Added in key prop with the label value (labels are unique).

**Review by solovevayaroslavna — APPROVED** [review]

**Review by solovevayaroslavna — APPROVED** [review]

**Pull request merged**